starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,181 @@
+package starportcmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/entrywriter"
+	"github.com/tendermint/starport/starport/pkg/yaml"
+	"github.com/tendermint/starport/starport/services/network"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+)
+
+type ShowType string
+
+const (
+	chainShowInfo     ShowType = "info"
+	chainShowGenesis  ShowType = "genesis"
+	chainShowAccounts ShowType = "accounts"
+	chainShowPeers    ShowType = "peers"
+)
+
+var (
+	showTypes = map[ShowType]struct{}{
+		chainShowInfo:     {},
+		chainShowGenesis:  {},
+		chainShowAccounts: {},
+		chainShowPeers:    {},
+	}
+
+	chainAccSummaryHeader = []string{"Genesis Account", "Coins"}
+)
+
+// NewNetworkChainShow creates a new chain show command to show
+// a chain on SPN.
+func NewNetworkChainShow() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "show [info|genesis|accounts|peers] [launch-id]",
+		Short: "Show details of a chain",
+		RunE:  networkChainShowHandler,
+		Args:  cobra.ExactArgs(2),
+	}
+
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	showType := ShowType(args[0])
+	if _, ok := showTypes[showType]; !ok {
+		cmd.Usage()
+		return fmt.Errorf("invalid arg %s", showType)
+	}
+
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID.
+	launchID, err := network.ParseLaunchID(args[1])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	c, err := nb.Chain(networkchain.SourceLaunch(chainLaunch))
+	if err != nil {
+		return err
+	}
+
+	content := ""
+	switch showType {
+	case chainShowGenesis:
+		content, err = formatChainGenesis(c)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), c, launchID)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(c network.Chain) (string, error) {
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return "", err
+	}
+	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
+		return "", fmt.Errorf("chain genesis not initialized: %s", genesisPath)
```

Comment:
The is not the initialization that should be run but the `prepare` process to construct the genesis

For UX I think it would be better to be able to run the command without the need to run prior command, so integrating the `(c Chain) Prepare` routine in the command. Also to make this command without side-effect (because it is a querying command) I would consider creating the genesis in a temporary repositoring and then print it to the console

This is open for debate

If this is a more complex problem. We can also implement `genesis` in a separate PR

Reply:
Great, this implementation will be bigger than this entire PR. I will implement another one. I only updated the messages now and added a TODO commend. I also opened issue #1904